pkg/virt-operator/install-strategy/create.go

```diff
@@ -0,0 +1,996 @@
+							log.Log.Errorf("Failed to delete service %+v: %v", cachedService, err)
+							return false, err
+						}
+
+						log.Log.V(2).Infof("service %v deleted. It must be re-created", cachedService.GetName())
+						return false, nil
+					}
+				}
+			} else {
+				// Patch if old version
+				var ops []string
+
+				// Add Labels and Annotations Patches
+				labelAnnotationPatch, err := createLabelsAndAnnotationsPatch(&service.ObjectMeta)
+				if err != nil {
+					return false, err
+				}
+				ops = append(ops, labelAnnotationPatch...)
+
+				_, err = core.Services(kv.Namespace).Patch(service.Name, types.JSONPatchType, generatePatchBytes(ops))
+				if err != nil {
+					return false, fmt.Errorf("unable to patch service %+v: %v", service, err)
+				}
+				log.Log.V(2).Infof("service %v updated", service.GetName())
+			}
+		} else {
+			log.Log.V(4).Infof("service %v is up-to-date", service.GetName())
+		}
+	}
+
+	// Add new SCC Privileges and remove unsed SCC Privileges
+	for _, sccPriv := range targetStrategy.customSCCPrivileges {
+		var curSccPriv *customSCCPrivilegedAccounts
+		if prevStrategy != nil {
+			for _, entry := range prevStrategy.customSCCPrivileges {
+				if sccPriv.TargetSCC == entry.TargetSCC {
+					curSccPriv = entry
+					break
+				}
+			}
+		}
+
+		privSCCObj, exists, err := stores.SCCCache.GetByKey(sccPriv.TargetSCC)
+		if !exists {
+			continue
+		} else if err != nil {
+			return false, err
+		}
+
+		privSCC, ok := privSCCObj.(*secv1.SecurityContextConstraints)
+		if !ok {
+			return false, fmt.Errorf("couldn't cast object to SecurityContextConstraints: %+v", privSCCObj)
+		}
+		privSCCCopy := privSCC.DeepCopy()
+
+		modified := false
+		users := privSCCCopy.Users
+
+		// remove users from previous
+		if curSccPriv != nil {
+			for _, acc := range curSccPriv.ServiceAccounts {
+				shouldRemove := true
+				// only remove if the target doesn't contain the same
+				// rule, otherwise leave as is.
+				for _, targetAcc := range sccPriv.ServiceAccounts {
+					if acc == targetAcc {
+						shouldRemove = false
+						break
+					}
+				}
+				if shouldRemove {
+					removed := false
+					users, removed = remove(users, acc)
+					modified = modified || removed
+				}
+			}
+		}
+
+		// add any users from target that don't already exist
+		for _, acc := range sccPriv.ServiceAccounts {
+			if !contains(users, acc) {
+				users = append(users, acc)
+				modified = true
+			}
+		}
+
+		if modified {
+			userBytes, err := json.Marshal(users)
+			if err != nil {
+				return false, err
+			}
+
+			data := []byte(fmt.Sprintf(`{"users": %s}`, userBytes))
+			_, err = scc.SecurityContextConstraints().Patch(sccPriv.TargetSCC, types.StrategicMergePatchType, data)
+			if err != nil {
+				return false, fmt.Errorf("unable to patch scc: %v", err)
+			}
+		}
+	}
+
+	if takeUpdatePath {
+		// UPDATE PATH IS
+		// 1. daemonsets - ensures all compute nodes are updated to handle new features
+		// 2. wait for daemonsets to roll over
```

Comment:
This comment says that you are waiting for the daemonsets before launching controllers but down below you create daemonsets then controllers and then wait for each.

Reply:
good catch. i'll update the comment to reflect reality. i changed some of this around after the fact.